lib/pengine/native.c

```diff
@@ -258,7 +259,8 @@ rsc_is_on_node(pe_resource_t *rsc, const pe_node_t *node, int flags)
                && (rsc->running_on == NULL)) {
         return true;
 
-    } else if (!pcmk_is_set(flags, pe_find_current) && rsc->allocated_to
+    } else if (!pcmk_is_set(flags, pcmk_rsc_match_current_node)
+               && (rsc->allocated_to != NULL)
```

Comment:
`pe__same_node(rsc->allocated_to, node)` if you feel like it

Reply:
yes but that will be a separate pr